packages/cli/src/util/input/project-selector.ts

```diff
@@ -0,0 +1,69 @@
+import { CheckboxChoiceOptions } from 'inquirer';
+import Client from '../client';
+import { RepoProjectsConfig } from '../link/repo';
+import {
+  getWorkspaces,
+  LocalFileSystemDetector,
+  getWorkspacePackagePaths,
+} from '@vercel/fs-detectors';
+import path from 'path';
+
+export async function projectSelector(
+  client: Client,
+  cwd: string,
+  repoConfig?: RepoProjectsConfig,
+  autoConfirm: boolean = false
+) {
+  require('./patch-inquirer');
+
+  const choices: Array<CheckboxChoiceOptions> = [];
+
+  if (repoConfig) {
+    for (const project of repoConfig.projects) {
+      choices.push({
+        name: project.name,
+        value: project.directory,
+        checked: true,
+      });
+    }
+  } else {
+    const localFS = new LocalFileSystemDetector(cwd);
+    const workspaces = await getWorkspaces({ fs: localFS });
+    const packagesPaths = await Promise.all(
+      workspaces.map(workspace =>
+        getWorkspacePackagePaths({ fs: localFS, workspace })
+      )
+    );
+    const packages = await Promise.all(
+      packagesPaths.flat().map(async packagePath => {
+        const packagePackageJSONBuffer = await localFS
+          .readFile(path.join(packagePath, 'package.json'))
+          .catch(() => null);
```

Comment:
**suggestion (soft-blocking):** debug log error

In this or a follow-up PR: Can we debug log any error that comes from this? I'm always skeptical of ignoring errors.